crates/block-producer/src/batch_builder/batch.rs

```diff
@@ -97,13 +96,13 @@ impl TransactionBatch {
                         continue;
                     }
 
-                    match found_unauthenticated_notes {
-                        Some(ref found_notes) => match found_notes.get(&input_note_header.id()) {
-                            Some(_path) => input_note.nullifier().into(),
-                            None => input_note.clone(),
-                        },
-                        None => input_note.clone(),
-                    }
+                    // If an unauthenticated note was found in the store, transform it to an authenticated one
+                    // (i.e. erase additional note details except the nullifier)
+                    found_unauthenticated_notes
```

Comment:
Nit: it feels weird having an optional container as input.

We can remove a layer of indirection by dropping the `Option<_container_>`. Callers can transform their optional container with `.unwrap_or_default` if necessary.

Reply:
I don't have strong opinion on this matter, but your suggestion can help to simplify code a bit, so I decided to get rid of `Option`, thank you!